api/src/main/java/org/openmrs/util/OpenmrsConstants.java

```diff
@@ -597,6 +602,12 @@ public static String getOpenmrsProperty(String property) {
 	 * Specifies the uuid of the concept which represents drug non coded
 	 */
 	public static final String GP_DRUG_ORDER_DRUG_OTHER = "drugOrder.drugOther";
+	
+	/**
+	 * Global property that stores the base url for the application.
+	 */
+	public static final String GP_HOST_URL = "host.url";
```

Comment:
do we really need this?

Reply:
How would you know the domain? I think you're assuming it MUST point back to our application which I doubt is always true, an implementation might choose to route it through some other service, they might be behind a proxy etc.